mixer.cpp

```diff
@@ -41,9 +41,61 @@ static uint8_t *convertMono8(SDL_AudioCVT *cvt, const uint8_t *data, int freq, i
 	return out;
 }
 
+static Mix_Chunk *loadAndConvertWav(SDL_RWops *rw, int freeRw, int freq, SDL_AudioFormat dstFormat, int dstChannels, int dstFreq)
+{
+	SDL_AudioSpec wavSpec;
+	Uint8 *wavBuf;
+	Uint32 wavLen;
+	SDL_AudioCVT cvt;
+
+	if (!SDL_LoadWAV_RW(rw, freeRw, &wavSpec, &wavBuf, &wavLen)) {
+		return 0;
+	}
+
+	if (wavSpec.freq == 22050) {
+		freq = (int)(freq * 2.21764064f);
+	} else if (wavSpec.freq == 44100) {
+		freq = (int)(freq * 4.43528128f);
+	}
```

Comment:
Assuming the data files only contain 9943, 22050 or 44100hz frequency wave files, this looks equivalent to freq *= (wavSpec.freq  / 9943f). Can this be simplified?

Reply:
This is how it was implemented in the original executable (in a more roundabout way) and I didn't check what frequency is in the data files. I'm not sure if your version is simpler, but it's more concise.
If the data files in all three versions (15th/20th/win3.1) don't contain other frequencies then it can be changed.

Reply:
I checked all three versions and there are 48khz and 11Khz files.

```
data_20th_linux/game/game/WGZ/original$ for f in *gz; do zcat $f|file -|cut -d, -f5; done|sort -u
 mono 22050 Hz
 mono 9943 Hz
data_20th_linux/game/game/WGZ$ for f in *gz; do zcat $f|file -|cut -d, -f5; done|sort -u
 mono 22050 Hz
 mono 44100 Hz
 mono 48000 Hz
 mono 9943 Hz
 stereo 22050 Hz
 stereo 44100 Hz

data_15th/Data/dlx$ for f in *wav; do file $f|cut -d, -f5; done|sort -u
 mono 22050 Hz
 mono 9943 Hz

data_15th/Data/dlx/rmsnd$ for f in *wav; do file $f|cut -d, -f5; done|sort -u
 mono 22050 Hz
 stereo 22050 Hz

aw_win31/out$ for f in *WAV; do file $f|cut -d, -f5; done|sort -u
 mono 11025 Hz
```

But I feel using the 9943 divisor makes the code clearer. What about changing it to something like:

```
if (wavfreq == 22050 || wavfreq == 44100) {
  freq = int(freq * (wavfreq / 9943))
}
```

Reply:
Maybe this can be extended to 48khz files, too? These files are played with several frequencies, including `period==20`, which is equal to 9943hz (as the 20th edition offset the table by 1).

```
file081-EX-1.wgz
/dev/stdin: RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 48000 Hz

16002_eau.asm:4DB0: (18) playSound(res=81, freq=20, vol=29, channel=1)
16002_eau.asm:530F: (18) playSound(res=81, freq=10, vol=40, channel=1)
16002_eau.asm:53B2: (18) playSound(res=81, freq=15, vol=40, channel=1)
16002_eau.asm:53F9: (18) playSound(res=81, freq=10, vol=40, channel=1)
```